samples/javascript_nodejs/21.corebot-app-insights/bots/dialogBot.js

```diff
@@ -11,41 +11,35 @@ class DialogBot extends ActivityHandler {
      */
     constructor(conversationState, userState, dialog) {
         super();
-
-        if (!conversationState) { throw new Error('[DialogBot]: Missing conversationState parameter.'); }
-        if (!userState) { throw new Error('[DialogBot]: Missing userState parameter.'); }
-        if (!dialog) { throw new Error('[DialogBot]: Missing dialog parameter.'); }
+        if (!conversationState) throw new Error('[DialogBot]: Missing parameter. conversationState is required');
+        if (!userState) throw new Error('[DialogBot]: Missing parameter. userState is required');
+        if (!dialog) throw new Error('[DialogBot]: Missing parameter. dialog is required');
 
         this.conversationState = conversationState;
         this.userState = userState;
         this.dialog = dialog;
+        this.dialogState = this.conversationState.createProperty('DialogState');
 
-        this.onTurn(async (turnContext, next) => {
-            await this.conversationState.saveChanges(turnContext, false);
-            await this.userState.saveChanges(turnContext, false);
-
-            // Ensure next BotHandler is executed.
-            await next();
-        });
-
-        this.onMessage(async (turnContext, next) => {
+        this.onMessage(async (context, next) => {
             console.log('Running dialog with Message Activity.');
-            const dialogState = this.conversationState.createProperty('DialogState');
 
-            await this.dialog.run(turnContext, dialogState);
+            // Run the Dialog with the new message Activity.
+            await this.dialog.run(context, this.dialogState);
 
-            // Ensure next BotHandler is executed.
+            // By calling next() you ensure that the next BotHandler is run.
             await next();
         });
+    }
 
-        this.onDialog(async (context, next) => {
-            // Save any state changes. The load happened during the execution of the Dialog.
-            await this.conversationState.saveChanges(context, false);
-            await this.userState.saveChanges(context, false);
+     /**
```

Comment:
Very minor - needs one less sapce to line up.

```suggestion
    /**
```

Reply:
Amended. Thanks.